engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/error/ErrorEventTest.java

```diff
@@ -619,4 +621,90 @@ public void shouldThrowErrorOnEndEventWithExpression() {
             tuple(BpmnElementType.BOUNDARY_EVENT, ProcessInstanceIntent.ELEMENT_COMPLETED),
             tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
   }
+
+  @Test
+  public void shouldPropagateErrorCodeVariable() {
+    // Given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b -> b.errorEventDefinition().error(ERROR_CODE).errorCodeVariable("errorCode"))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(JOB_TYPE)
+        .withErrorCode(ERROR_CODE)
+        .throwError();
+
+    // Then
+    final long scopeKey =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.BOUNDARY_EVENT)
+            .getFirst()
+            .getKey();
+
+    final Record<VariableRecordValue> variableRecords =
+        RecordingExporter.variableRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withScopeKey(scopeKey)
+            .withName("errorCode")
+            .getFirst();
+    Assertions.assertThat(variableRecords.getValue()).hasValue("\"ERROR\"");
+  }
+
+  @Test
+  public void shouldPropagateErrorMessageVariable() {
+    // Given
+    final var process =
+        Bpmn.createExecutableProcess(PROCESS_ID)
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+            .boundaryEvent(
+                "error-boundary-event",
+                b ->
+                    b.errorEventDefinition().error(ERROR_CODE).errorMessageVariable("errorMessage"))
+            .endEvent()
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    // when
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(JOB_TYPE)
+        .withErrorCode(ERROR_CODE)
+        .withErrorMessage("error-message")
+        .throwError();
```

Comment:
👍 This is a great test

❌ I'd like to see 2 more test cases added:

- [ ] verify what happens when `errorMessageVariable` is specified, but no `errorMessage` is passed along with the thrown error. I expect that we create the variable with the value `null` or `""` (empty string)
- [ ] verify what happens when an Error End Event is used to throw an error and caught in an Error Boundary Event that wants to create a variable out of the error message. Again I expect that we create the variable with the name specified in the `errorMessageVariable` with value either `null` or `""` (empty string)